### ISSUE TITLE
Enhancement: Use "invitation code" to verify

### ISSUE THREAD

**Issue by vicevolf** [danny-avila/LibreChat#3227]
### What features would you like to see added?

In my opinion, one of the more effective ways to protect public community services is to use "invitation codes" to verify actions such as the registration of new users.

### More details

Examples are as follows:

1. Optional mode, new users must enter an invitation code to register;
2. Optional feature, old users (e.g., one week after first using the service) can get a new invitation code each week (up to 3 codes) to invite new users;
3. Optional feature, another set of invitation codes can provide advanced features, such as users being able to exchange for token balances;

This strategy is inspired by the early registration scheme of the Blue Sky Community, which can protect service security and community ecology, and limit the expansion speed of the service. Additionally, the invitation code has the potential to integrate with other services. Accordingly, there might also be a need to manage the invitation codes, such as importing invitation codes, tracking their usage, and invalidating codes.

### Which components are impacted by your request?

General

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by VincentJGeisler** [comment]
yes this! a way of preventing open signups is badly needed.

**Comment by berry-13** [comment]
what about something like this?
#3012

**Comment by vicevolf** [comment]
> 像这样的事情怎么样？ [#3012 ](https://github.com/danny-avila/LibreChat/pull/3012)

Not entirely similar, this is an additional method of security verification, allowing relatively automatic circulation within small social circles, rather than a way to invite new users.

**Comment by quark-zju** [comment]
If you use nginx, you might configure it like this to only allow registration if an "invite" URL was accessed.

```nginx
server {
        listen 443 ssl;

        server_name ...;
        ssl_certificate ...;
        ssl_certificate_key ...;

        autoindex off;
        client_max_body_size 20M;

        location = /SECRET_INVITE_URL {
                add_header Set-Cookie "invite=SECRET; Path=/api/auth/register; HttpOnly";
                add_header Set-Cookie "invite=SECRET; Path=/register; HttpOnly";
                return 302 /;
        }

        location / {
                try_files "" @backend;
        }

        location /register {
                try_files "" @check_cookie;
        }

        location /api/auth/register {
                limit_except POST {
                        deny all;
                }
                try_files "" @check_cookie;
        }

        location @check_cookie {
                if ($cookie_invite != "SECRET") {
                        add_header Content-Type text/plain always;
                        return 403 "Access Denied";
                }
                try_files "" @backend;
        }

        location @backend {
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                proxy_set_header Host $host;
                proxy_pass http://127.0.0.1:3080;
                proxy_http_version 1.1;
                proxy_set_header Upgrade $http_upgrade;
                proxy_set_header Connection "upgrade";
                proxy_buffering off;
        }
}
```

**Comment by danny-avila** [comment]
Closed by #3012 

> Optional mode, new users must enter an invitation code to register;
This is the only one in scope of the project, the other requested features seem too niche

>  Optional feature, old users (e.g., one week after first using the service) can get a new invitation code each week (up to 3 codes) to invite new users;
> Optional feature, another set of invitation codes can provide advanced features, such as users being able to exchange for token balances;
